sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,248 @@
+
+    @property
+    @abstractmethod
+    def shortening_velocity(self) -> Expr:
+        """An expression representing the pathway's shortening velocity."""
+        pass
+
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}(attachments={self.attachments})'
+
+
+class LinearPathway(PathwayBase):
+    """Linear pathway between a pair of attachment points.
+
+    Explanation
+    ===========
+
+    A linear pathway forms a straight-line segment between two points and is
+    the simplest pathway that can be formed. It will not interact with any
+    other objects in the system, i.e. a ``LinearPathway`` will interest other
+    objects to ensure that the path between its two ends (its attachments) is
+    the shortest possible.
+
+    Examples
+    ========
+
+    As the ``_pathway.py`` module is experimental, it is not yet part of the
+    ``sympy.physics.mechanics`` namespace. ``LinearPathway`` must therefore be
+    imported directly from the ``sympy.physics.mechanics._pathway`` module.
+
+    >>> from sympy.physics.mechanics._pathway import LinearPathway
+
+    To construct a pathway, two points are required to be passed to the
+    ``attachments`` parameter as a ``tuple``.
+
+    >>> from sympy.physics.mechanics import Point
+    >>> attachments = (Point('pO'), Point('pI'))
+    >>> linear_pathway = LinearPathway(attachments)
+    >>> linear_pathway
+    LinearPathway(attachments=(pO, pI))
+
+    The pathway create above isn't very interesting without the positions and
+    velocities of its attachment points being described. Without this its not
+    possible to describe how the pathway moves, i.e. its length or its
+    shortening velocity.
+
+    >>> from sympy.physics.mechanics import ReferenceFrame
+    >>> from sympy.physics.vector import dynamicsymbols
+    >>> N = ReferenceFrame('N')
+    >>> q = dynamicsymbols('q')
+    >>> attachments[-1].set_pos(attachments[0], q * N.x)
+    >>> attachments[-1].pos_from(attachments[0])
+    q(t)*N.x
+
+    A pathway's length can be accessed via its ``length`` attribute.
+
+    >>> linear_pathway.length
+    sqrt(q(t)**2)
+
+    Note how what appears to be an overly-complex expression is returned. This
+    is actually required as it ensures that a pathway's length is always
+    positive.
+
+    A pathway's shortening velocity can be accessed similarly via its
+    ``shortening_velocity`` attribute.
+
+    >>> linear_pathway.shortening_velocity
+    -q(t)*Derivative(q(t), t)/sqrt(q(t)**2)
+
+    Parameters
+    ==========
+
+    attachments : tuple[Point, Point]
+        The pair of ``Point`` objects between which the linear pathway spans.
+
+    """
+
+    def __init__(
+        self,
+        attachments: tuple[Point, ...],
+    ) -> None:
+        """Initializer for ``LinearPathway``.
+
+        Parameters
+        ==========
+
+        attachments : tuple[Point, Point]
+            The pair of ``Point`` objects between which the linear pathway
+            spans.
```

Comment:
If there are only two points, I'd recommend `LinearPathway(P1, P2)` as the api.

Reply:
Addressed in https://github.com/sympy/sympy/pull/24914/commits/b44a2497ee3fd0debdc0978cb26db31842b258e6.